src/actions.rs

```diff
@@ -35,6 +35,7 @@ pub struct InnerAction {
     pub section_members: BTreeMap<SectionInfo, Vec<Node>>,
 
     pub merge_infos: Option<MergeInfo>,
+    pub merge_needed: bool,
```

Comment:
Could you add one test case that set it to true (on top of the existing ones that have false).?

Reply:
Given that it's currently mocked, wouldn't such a test set the bool and then assert that it's actually set?

Reply:
I mean, currently since it's mocked there is no path in the flows that triggers it